### ISSUE TITLE
Recieve Instantiation Error on Argument-Matching Arrays

### ISSUE THREAD

**Issue by tinyrye** [oleksiyp/mockk#12]
I could not see in the documentation which I find straightforward any mention of not supporting Array argument matching.

With that in mind could I asking whether this is a Mockk bug, or what is being done wrong in this code.

Stacktrac
```
java.lang.InstantiationError: [Lorg.springframework.jdbc.core.namedparam.SqlParameterSource;
	at sun.reflect.GeneratedSerializationConstructorAccessor19.newInstance(Unknown Source)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at io.mockk.shadowed.objenesis.instantiator.sun.SunReflectionFactoryInstantiator.newInstance(SunReflectionFactoryInstantiator.java:48)
	at io.mockk.proxy.MockKProxyMaker.newEmptyInstance(MockKProxyMaker.java:184)
	at io.mockk.proxy.MockKProxyMaker.instance(MockKProxyMaker.java:52)
	at io.mockk.impl.InstantiatorImpl.instantiate(Instantiator.kt:53)
	at io.mockk.impl.InstantiatorImpl.signatureValue(Instantiator.kt:109)
	at io.mockk.impl.CallRecorderImpl.matcher(CallRecording.kt:97)
```

What I am trying to mock:
`org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate.batchUpdate(String, Array<SqlParameterSource>)`

How I am trying to mock it:
```
        every {
            localDb.batchUpdate(
                match{ sql: String ->
                    sql == PaymentDao.Companion.PAYMENT_INSERT_SQL_STMT
                },
                any<Array<SqlParameterSource>>()
            )
         } returns intArrayOf(1)
```

**Comment by oleksiyp** [comment]
I think its a bug. I was fixing similiar issue in master, will test your
case anyway. So should be okay in new release. I'll try to release it soon.

There is really a lot of changes there. So need some time for testing it.
Mostly it is internal refactoring, but as well experimental support of
Kotlin JS.

18 дек. 2017 г. 5:25 PM пользователь "tinyrye" <notifications@github.com>
написал:

**Comment by tinyrye** [comment]
Thank you; we are still slowly in our Kotlin stack adding Mockk, and the more stable the new release is as a whole the better for all of us users.

**Comment by oleksiyp** [comment]
Released :-) Please try release 1.6. It should fix your case. As I said it can be not so stable as desired due to big number of changes, from other side I covered code with many Unit tests (from 69% to 80%), so may be it's not so bad. Please report any bugs. If your case is fixed close the ticket

**Comment by tinyrye** [comment]
Thank you, sir!  The error resolved and I see no regressions currently.
